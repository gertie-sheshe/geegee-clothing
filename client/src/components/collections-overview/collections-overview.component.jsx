import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import CollectionPreview from '../collection-preview/collection-preview.component';

import { CollectionsOverviewContainer } from './collections-overview.styles';

// Selectors
import { selectCollectionsForPreview } from '../../redux/shop/shop.selectors';

export const CollectionsOverview = ({ collections }) => {
  return (
    <CollectionsOverviewContainer>
      {collections.map(({ id, ...collection }) => (
        <CollectionPreview key={id} {...collection} />
      ))}
    </CollectionsOverviewContainer>
  );
};

const mapStateToProps = createStructuredSelector({
  collections: selectCollectionsForPreview,
});

export default connect(mapStateToProps)(CollectionsOverview);
