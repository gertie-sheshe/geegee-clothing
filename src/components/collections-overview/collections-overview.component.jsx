import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import CollectionPreview from '../collection-preview/collection-preview.component';

import { CollectionsOverviewContainer } from './collections-overview.styles';

// Selectors
import { selectCollectionsForPreview } from '../../redux/shop/shop.selectors';

const CollectionsOverview = ({ collections }) => (
  <CollectionsOverviewContainer>
    {collections.map(({ id, ...collection }) => (
      <CollectionPreview key={id} {...collection} />
    ))}
  </CollectionsOverviewContainer>
);

const mapStateToProps = createStructuredSelector({
  collections: selectCollectionsForPreview,
});

export default connect(mapStateToProps)(CollectionsOverview);
