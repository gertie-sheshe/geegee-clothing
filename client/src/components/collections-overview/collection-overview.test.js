import React from 'react';
import { shallow } from 'enzyme';

import { CollectionsOverview } from './collections-overview.component';

describe('Collection Overview Component', () => {
  it('Should render Collection Overview component', () => {
    let wrapper = shallow(<CollectionsOverview collections={[]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
