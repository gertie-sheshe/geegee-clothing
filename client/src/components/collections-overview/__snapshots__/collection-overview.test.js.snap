// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collection Overview Component Should render Collection Overview component 1`] = `ShallowWrapper {}`;
